auth_cas/wizard/res_config.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+# © 2016 Roméo Guillot Roméo Guillot (http://www.opensource-elanz.fr).
+# © 2016-2018 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import logging
+from odoo.exceptions import ValidationError
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+from odoo.tools.translate import _
+from urlparse import urlparse
+
+_logger = logging.getLogger(__name__)
+try:
+    from ..pycas import login
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+default_host = 'https://localhost'
+default_port = 8443
+
+
+class CasBaseConfigSettings(models.TransientModel):
+    """
+    The fields declared here are used
+    to manage settings of the CAS server.
+    """
+
+    _inherit = 'base.config.settings'
+    cas_activated = fields.Boolean(
+        'CAS authentication activated',
+        help='The CAS authentication only works if you are in a single'
+             ' database mode. You can launch the Odoo Server with the'
+             ' option --db-filter=YOUR_DATABASE to do so.')
+    cas_server = fields.Char('CAS Server address', size=64)
+    cas_server_port = fields.Integer('CAS Server port')
+    cas_create_user = fields.Boolean(
+        'Users created on the fly',
+        help='Automatically create local user accounts for'
+             ' new users authenticating via CAS', default=True)
+
+    # Getter is required for fields stored in base.config.settings
+    @api.model
+    def get_default_cas_values(self, fields):
+        icp = self.env['ir.config_parameter']
+        return {
+            'cas_activated': safe_eval(
+                icp.get_param('cas_auth.cas_activated', 'False')),
+            'cas_server': icp.get_param('cas_auth.cas_server', default_host),
+            'cas_server_port': int(
+                icp.get_param('cas_auth.cas_server_port', default_port)),
+            'cas_create_user': safe_eval(
+                icp.get_param('cas_auth.cas_create_user', 'True')),
+        }
+
+    # Setter is required too
+    @api.multi
+    def set_cas_values(self):
+        self.ensure_one()
+        icp = self.env['ir.config_parameter']
+
+        error = True
+
+        # If the host OR the port is valid
+        if self.cas_server or self.cas_server_port:
```

Comment:
What is the point to check if any of them are valid?
Right after is we required that both are valid

Reply:
yes right I am improving code